utils/panel/panel.go

```diff
@@ -0,0 +1,173 @@
+package panel
+
+import (
+	"errors"
+
+	"github.com/mattermost/mattermost-plugin-api/utils/bot/logger"
+	"github.com/mattermost/mattermost-plugin-api/utils/bot/poster"
+	"github.com/mattermost/mattermost-plugin-api/utils/common"
+	"github.com/mattermost/mattermost-plugin-api/utils/panel/settings"
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+type Panel interface {
+	Set(userID, settingID string, value interface{}) error
+	Print(userID string)
+	ToPost(userID string) (*model.Post, error)
+	Clear(userID string) error
+	URL() string
+	GetSettingIDs() []string
+}
+
+type panel struct {
+	settings       map[string]settings.Setting
+	settingKeys    []string
+	poster         poster.Poster
+	logger         logger.Logger
+	store          PanelStore
```

Comment:
Implementation of `PanelStore`? Will that vary for different usecases? Isn't is just a KV store CRUD?